app/components/bookings/create/create-booking.component.ts

```diff
@@ -1,9 +1,124 @@
+import {BookingSession, Booking} from "../../../models/booking/booking";
+import Moment = moment.Moment;
+import {Response} from "@angular/http";
+import {BookingOverviewComponent} from "../overview/booking-overview.component";
 
 @Component({
   templateUrl: 'build/components/bookings/create/create-booking.component.html'
 })
 export class CreateBookingComponent {
-  constructor() {
+  private _workpackage: Workpackage;
+  private _bookingSession: BookingSession;
+  private _effortControl : AbstractControl;
+  private _descriptionControl: AbstractControl;
+  private _dateControl: AbstractControl;
+  private _newEtcControl: AbstractControl;
+  private _authForm: ControlGroup;
+  private _maxDate = moment().startOf('day').format('YYYY-MM-DD');
+  private _Date = moment().startOf('day').format('YYYY-MM-DD');
+  private _effort = moment().startOf('day').format('HH:mm');
+  private _Etc: number;
+
+  constructor(navParams: NavParams,
+              private _bookingsService: BookingService,
+              private _bookingSessionService: BookingSessionService,
+              private _formBuilder: FormBuilder,
+              private _navController: NavController) {
+
+    this._workpackage = navParams.get('workpackage');
+    this._bookingSession = navParams.get('bookingSession');
+
+    if(this._bookingSession) {
+      this._Date = moment(this._bookingSession.startTime).startOf('day').format('YYYY-MM-DD');
+      this._effort = this._calcEffortForSession(this._bookingSession.startTime, moment().unix());
+      this.refreshEtc();
+    }
+
+    this._authForm = _formBuilder.group({
+      '_descriptionControl': ['', Validators.compose([Validators.required, Validators.minLength(1)])],
+      '_dateControl': ['', Validators.required],
+      '_effortControl': ['', Validators.required]
+    });
+
+    this._newEtcControl = this._authForm.controls['_newEtcControl'];
+    this._effortControl = this._authForm.controls['_effortControl'];
+    this._descriptionControl = this._authForm.controls['_descriptionControl'];
+    this._dateControl = this._authForm.controls['_dateControl'];
+    this._Etc = this._workpackage.etc;
+  }
+
+  onSubmit() {
+    if (this._checkFieldsSet()) {
+      var booking: Booking = new Booking();
+      booking.date = moment(this._dateControl.value);
+      booking.description = this._descriptionControl.value;
+      booking.effort = this._stringToWorkday(this._effortControl.value);
+      booking.workpackage = this._workpackage;
+      this._bookingsService.create(booking, this._Etc).subscribe((booking: Booking) => {
+        if(this._bookingSession) {
+          //if there is a bookingsession, delete it when the booking was successful
+          this._bookingSessionService.delete(this._bookingSession).subscribe((response: Response) => {
+
+          });
+        }
+        //TODO: show ToastMsg Booking successful
+        this._navigateToBookingOverview();
```

Comment:
redirecting and showing the Toast should done, after the `BookingSession` has successfully been deleted, in case it gets deleted, since errors could occur on this request too.